Zend/zend_vm_def.h

```diff
@@ -8022,9 +8014,30 @@ ZEND_VM_HANDLER(182, ZEND_BIND_LEXICAL, TMP, CV, REF)
 		ZVAL_DEREF(var);
 		Z_TRY_ADDREF_P(var);
 	}
+	zend_closure_bind_var(closure, opline->result.num, var);
+	ZEND_VM_NEXT_OPCODE();
+}
+
+ZEND_VM_HANDLER(183, ZEND_FETCH_STATIC, CV, UNUSED, REF)
+{
+	USE_OPLINE
+	zval *var = GET_OP1_ZVAL_PTR(BP_VAR_W);
+	zval *val = &EX(func)->op_array.static_variables[opline->op2.num];
+
+	if (UNEXPECTED(Z_CONSTANT_P(val))) {
+		SAVE_OPLINE();
+		if (UNEXPECTED(zval_update_constant_ex(val, 1, NULL) != SUCCESS)) {
+			HANDLE_EXCEPTION();
+		}
+	}
+
+	if (opline->extended_value) {
+		ZVAL_MAKE_REF(val);
+	} 
+
+	zval_ptr_dtor(var);
+	ZVAL_COPY(var, val);
```

Comment:
In general, "val" may be a part of "var" and destroying "var" before incrementing "val", may cause problems. (may be I'm wrong).